src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -0,0 +1,122 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useMemo, useState} from 'react';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Icon from '@components/Icon';
+import * as Expensicons from '@components/Icon/Expensicons';
+import * as Illustrations from '@components/Icon/Illustrations';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import type {CentralPaneNavigatorParamList} from '@navigation/types';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+import withPolicyAccessOrNotFound from './withPolicyAccessOrNotFound';
+import type {WithWorkspaceAccessProps} from './withPolicyAccessOrNotFound';
+
+type PolicyForList = {
+    value: string;
+    text: string;
+    keyForList: string;
+    isSelected: boolean;
+    rightElement: React.ReactNode;
+};
+
+type WorkspaceCategoriesOnyxProps = {
+    /** Collection of categories attached to a policy */
+    policyCategories: OnyxEntry<OnyxTypes.PolicyCategories>;
+};
+
+type WorkspaceCategoriesPageProps = WorkspaceCategoriesOnyxProps & WithWorkspaceAccessProps & StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.WORKSPACE.CATEGORIES>;
+
+function WorkspaceCategoriesPage({policyCategories}: WorkspaceCategoriesPageProps) {
```

Comment:
That means we probably need to add some redirect here/ show 404 when the policy is not Paid Group policy.

The paid policy that this shows up to needs to have the `policy?.isPolicyExpenseChatEnabled` set to true

Reply:
>The paid policy that this shows up to needs to have the policy?.isPolicyExpenseChatEnabled set to true

For my own knowledge, why do we need to do this here? Isn't it enough that the overall workspace settings don't show for a paid plan without that. Like, why do we have to add it to each and every page (knowing we'll then need to remove it later)?

Reply:
> Isn't it enough that the overall workspace settings don't show for a paid plan without that. Like, why do we have to add it to each and every page (knowing we'll then need to remove it later)?

Well I think there will be some Higher Order Component wrapping all the toggle-able pages, but essentially even if you do not have this button in UI you can put the categories URL with policyID of a free policy in and then we need some logic to check that this page is not allowed for free policy

Reply:
> you can put the categories URL with policyID of a free policy in and then we need some logic to check that this page is not allowed for free policy

Ah, fair enough.. good point. When `isPolicyExpenseChatEnabled` goes away, what happens? Wouldn't using the policy type be better and future proof?

Reply:
I've updated HOC that checks `isPolicyExpenseChatEnabled` and redirects 404 for non-paid group policies

Reply:
> When isPolicyExpenseChatEnabled goes away, what happens? Wouldn't using the policy type be better and future proof?

We will keep using it until migration is done, then we can remove it safely